Added three tutorial sections to the Qiita documentation (#3032)
biocore/qiita: qiita_pet/support_files/doc/source/reanalysis/reanalysis.rst

.. code-block:: python

    # save the filtered feature table and visualisations
    sample_filtered_data.filtered_table.save('./results/sample-filtered-table.qza')
    vis_sample_filtered.visualization.save('./results/sample-filtered-table.qzv')
    feature_filtered_data.filtered_table.save('./results/feature-filtered-table.qza')
    vis_feature_filtered.visualization.save('./results/feature-filtered-table.qzv')

    # for future reference save the ids of the samples that met filtering criteria
    filtered_table = feature_filtered_data.filtered_table.view(q.Metadata)
    filtered_ids = "\n".join(filtered_table.to_dataframe().index) + "\n"
    with open('./results/filtered.ids', 'w') as f:

>     f.write(filtered_ids)

Reviewer: ```suggestion
        f.write(filtered_ids)
```

You have now obtained a filtered dataset, but will need to classify the features if you want to use analyses that take into account phylogenetic distance. Therefore we need to extract sequences from the dataset and ‘insert' them into a reference phylogenetic tree (this placement identifies their taxonomic position). To extract representative sequences from the features we will use the ``feature_filtered_data`` (a frequency feature table) to make a fasta file (a universal DNA/protein sequence file format) of sequence feature data. You can view this file with ``less <file>`` or ``cat <file> | head``, as it is a plain text file. Often such files have information about the origin of the sequences (on the > line before the sequence), but we will use the sequence itself as an ID as the taxonomic data is currently unknown. Fragment insertion, mapping representative sequences from the samples to a reference database will allow taxonomic classification of the ASVs. First extract representative sequences from the data:

.. code-block:: python

    with open('./results/sequences.fna', 'w') as f:

>     seqs = ''
>     for i,seq in enumerate(feature_filtered_data.filtered_table.view(pd.DataFrame).columns):
>         seqs = seqs + '>' + seq + '\n' + seq + '\n'
>     f.write(seqs[:-1])

Reviewer: ```suggestion
        for seq in feature_filtered_data.filtered_table.view(pd.DataFrame).columns:
            f.write('>%s\n%s\n' % (seq, seq))
```

    # Load the Greengenes sequences and taxonomy and Clawback human stool weights
    human_stool_weights = q.Artifact.load('./reference/human-stool.qza')
    ref_seqs_v4 = q.Artifact.load('./reference/ref-seqs-v4.qza')
    ref_tax = q.Artifact.load('./reference/ref-tax.qza')

We can then generate a classifier and classify the sequences:

.. code-block:: python

    # Generate a classifier with the loaded reference data
    human_stool_v4_result = feature_classifier.methods.fit_classifier_naive_bayes(reference_reads=ref_seqs_v4, reference_taxonomy=ref_tax, class_weight=human_stool_weights)

    # Use classifier to classify sequences
    bespoke_taxonomy = feature_classifier.methods.classify_sklearn(reads=rep_seqs, classifier=human_stool_v4_result.classifier, n_jobs=-1)

The classifier training will give a warning
"`The TaxonomicClassifier artifact that results from this method was trained using scikit-learn version 0.23.1. It cannot be used with other versions of scikit-learn.`"
You can check your version of scikit-learn by typing the following in your python console:

.. code-block:: python

>     print('The scikit-learn version is {}.'.format(sklearn.__version__))

Reviewer: ```suggestion
    import sklearn
    print('The scikit-learn version is {}.'.format(sklearn.__version__))
```
